portality/tasks/preservation.py

```diff
@@ -0,0 +1,589 @@
+        """
+        if response.status_code == 200:
+            res_json = json.loads(response.text)
+            files = res_json["files"]
+            # Success response
+            # {"files": [{"name": "name_of_tarball.tar.gz",
+            #             "sha256": "decafbad"}]}
+            if files:
+                # Check if the response is type dict or list
+                if isinstance(files, dict):
+                    res_filename = files["name"]
+                    res_shasum = files["sha256"]
+                elif isinstance(files, list):
+                    if len(files) > 0:
+                        res_filename = files[0]["name"]
+                        res_shasum = files[0]["sha256"]
+
+                if res_filename and res_filename == tar_file:
+                    if res_shasum and res_shasum == sha256:
+                        app.logger.info("succesfully uploaded")
+                        model.uploaded_to_ia()
+                    else:
+                        model.failed("shasum in response doesn't match")
+                else:
+                    model.failed("tar filename in response doesn't match")
+
+            else:
+                # Error response
+                # {"result": "ERROR","manifest_type": "BagIt",
+                #     "manifests": [
+                #         {"id": "033168cd016a49eb8c3097d800f1b85f",
+                #             "result": "SUCCESS"},
+                #         {"id": "00003741594643f4996e2555a01e03c7",
+                #             "result": "ERROR",
+                #             "errors": [
+                #                   "missing_files": [],
+                #                   "mismatch_hashes": [{
+                #                       "file": "path/to/file",
+                #                       "expected": "decafbad",
+                #                       "actual": "deadbeaf"}],
+                #                   "manifest_parsing_errors": [
+                #                   "some weird error"]]}]}
+                result = res_json["result"]
+                if result and result == "ERROR":
+                    error_str = "Upload failed due error at IA server side"
+                else:
+                    error_str = "Unknown Error: Not a valid response"
+
+                app.logger.error(error_str)
+                model.failed(error_str)
+
+            model.save()
+        else:
+            app.logger.error(f"Upload failed {response.text}")
+            model.failed(response.text)
+            model.save()
+
+
+    @classmethod
+    def submit(cls, background_job):
+        """~~-> SubmitJob:Feature~~
+        Submit Background job"""
+        background_job.save(blocking=True)
+        preserve.schedule(args=(background_job.id,), delay=10)
+
+@main_queue.task(**configure("preserve"))
+@write_required(script=True)
+def preserve(job_id):
+    """~~-> CreateBackgroundTask:Feature"""
```

Comment:
Same as above, I think this can be omitted.  If you felt that this one is particularly important to identify, maybe we would annotate it as a queue entry point, like `PreservationBackgroundTask:Queue`